### ISSUE TITLE
Changing from openstack user_credentials to application_credentials in metakube_cluster resource results in both showing in state

### ISSUE THREAD

**Issue by MarleDK** [syseleven/terraform-provider-metakube#70]
Hi,

I have an error on version 5.0.3, where I can't change from `user_credentials` to `application_credentials`, and I am now stuck, not even being able to revert to `user_credentials` either.

I tried using application_credentials instead of user_credentials in my openstack block:
```
@@ -80,10 +80,9 @@ resource "metakube_cluster" "this" {
     }
     cloud {
       openstack {
-        user_credentials {
-          username     = data.openstack_identity_auth_scope_v3.this.user_name
-          project_id   = data.openstack_identity_auth_scope_v3.this.project_id
-          project_name = data.openstack_identity_auth_scope_v3.this.project_name
+        application_credentials {
+          id     = "some_id"
+          secret = "some_secret"
         }
```

But when I tried to apply this, I get this error:
```
│ Error: Please use only one of either user_credentials or application_credentials
│
│   with metakube_cluster.this,
│   on main.tf line 82, in resource "metakube_cluster" "this":
│   82:       openstack {
```

And when looking at the state, I can see why, it doesn't remove the user_credentials:
```
       cloud {

            openstack {
                ...
                application_credentials {
                    id     = "some_id"
                    secret = (sensitive value)
                }

                user_credentials {
                    password     = (sensitive value)
                    project_id   = "some_project_id"
                    project_name = "our-stack"
                    username     = (sensitive value)
                }
            }
        }

```

I hope you can help me resolving this issue.

**Comment by furkhat** [comment]
Hi @MarleDK, this was fixed in `v5.0.5`

**Comment by MarleDK** [comment]
Thanks, it worked great for me now :)
